lib/puma/server.rb

```diff
@@ -439,6 +436,12 @@ def process_client(client, buffer)
         rescue StandardError => e
           @events.unknown_error self, e, "Client"
         end
+
+        if @options[:out_of_band]
+          @thread_pool.with_mutex do
```

Comment:
Does this actually create a mutex now on `thread_pool`? The `OOB` hook is executed within a mutex, rather than dispatched to be executed.

This also, connected with a test `#test_out_of_band_stream` creates a very strong dependency on execution:
the test assumes that all connections are in fact processed concurrently, but this might not always be true, because should we assume that `threads=1`, kind of similar to executing `Puma` with `threads=1`. The test does fail in this condition, as this is being executed on down failing edge of the event.